### PR TITLE
Documentation of test macros

### PR DESCRIPTION
**Pull request by tjfulle**

### CODE REVIEW

File: src/utils/src/FortranTestMacros.h [hunk 1]
```diff
@@ -9,6 +9,63 @@
...
+!
+!   The SETUP_TEST macro initializes module variables in the fortest module and
+!   initializes MPI for MPI builds of ForTrilinos (meaning that MPI should not
+!   be initialized elsewhere in the test!)
+!
+! - Adding and running subtests.  Use the macro ADD_SUBTEST_AND_RUN:
+!
+!      ADD_SUBTEST_AND_RUN(TestName)
+!
+!   where TestName is the name of a test defined in the program unit.  The test
```
Comment (aprokop) [inline src/utils/src/FortranTestMacros.h]: Do we require that the test name be a valid Fortran variable name? If yes, put some comment about this.

File: src/utils/src/FortranTestMacros.h [hunk 2]
```diff
@@ -9,6 +9,63 @@
...
+!   should be defined using the FORTRILINOS_UNIT_TEST and
+!   END_FORTRILINOS_UNIT_TEST macros (after the program's "contains" directive):
+!
+!      FORTRILINOS_UNIT_TEST(TestName)
+!        ! Test definition
+!      END_FORTRILINOS_UNIT_TEST(TestName)
+!
+!   These macros define a subroutine named "TestName" and declare an in/out
+!   logical "SUCCESS" that can be set to .FALSE. if the test fails.
```
Comment (aprokop) [inline src/utils/src/FortranTestMacros.h]: Put some wording about local vs global success when run with MPI.

**Comment by tjfulle** [comment]
@aprokop, I guess it’s implied in the statement that the macro creates a subroutine as the same name as the test.  I can add something more explicit.

**Comment by aprokop** [comment]
Did not notice that sentence later.

**Pull request merged**